src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1183 @@
+  }
+
+  /**
+   * Compile a Java file and return the compilation exit value.
+   *
+   * @param file the file to be compiled and executed
+   * @param classpath dependencies and complete classpath to compile and run the Java program
+   * @param packageName the package that the Java file is in
+   * @param timeoutLimit number of seconds allowed for the whole test suite to run
+   * @return exit value of compiling the Java file
+   */
+  private static int compileJavaFile(
+      File file, String classpath, String packageName, int timeoutLimit) {
+    // Obtain directory to carry out compilation and execution step.
+    File executionDir = getExecutionDirectory(file, packageName);
+
+    // Command to compile the input Java file.
+    String command = "javac -classpath " + systemClassPath;
+    // Add current directory to class path.
+    command += pathSeparator + ".";
+    if (classpath != null) {
+      // Add specified classpath to command.
+      command += pathSeparator + classpath;
+    }
+
+    command += " " + file.getAbsolutePath();
+
+    // Compile specified Java file.
+    return runProcess(command, executionDir, timeoutLimit).exitValue;
+  }
+
+  /**
+   * Run a Java file and return the standard output.
+   *
+   * @param file the file to be compiled and executed
+   * @param classpath dependencies and complete classpath to compile and run the Java program
+   * @param packageName the package that the Java file is in
+   * @param timeoutLimit number of seconds allowed for the whole test suite to run
+   * @return the standard output from running the Java file
+   */
+  private static String runJavaFile(
+      File file, String classpath, String packageName, int timeoutLimit) {
+    // Obtain directory to carry out compilation and execution step.
+    File executionDir = getExecutionDirectory(file, packageName);
+
+    // Directory path for the classpath.
+    String dirPath = null;
+    if (file.getParentFile() != null) {
+      dirPath = file.getParentFile().getPath();
+    }
+
+    // Fully-qualified classname.
+    String fqClassName = FilenameUtils.removeExtension(file.getName());
+    if (packageName != null) {
+      fqClassName = packageName + "." + fqClassName;
+    }
+
+    // Command to run the Java file.
+    String command = "java -classpath " + systemClassPath;
+    // Add current directory to class path.
+    command += pathSeparator + ".";
+    if (dirPath != null) {
+      // Use directory as defined by the file path.
+      command += pathSeparator + dirPath;
+    }
+    if (classpath != null) {
+      // Add specified classpath to command.
+      command += pathSeparator + classpath;
+    }
+    // Add JUnitCore command.
+    command += " org.junit.runner.JUnitCore " + fqClassName;
+
+    // Run the specified Java file.
+    return runProcess(command, executionDir, timeoutLimit).stdout;
+  }
+
+  /**
+   * Get directory to execute command in given file path and package name.
```

Comment:
It's good to say how the method is used -- this is important.  Thanks.  
Please also give the method's specification.  What does it return?  Example input and output could help here if you find it more concise or understandable.